dd-sdk-android-ktx/src/main/kotlin/com/datadog/android/ktx/coroutine/FlowExt.kt

```diff
@@ -0,0 +1,37 @@
+/*
+ * Unless explicitly stated otherwise all files in this repository are licensed under the Apache License Version 2.0.
+ * This product includes software developed at Datadog (https://www.datadoghq.com/).
+ * Copyright 2016-Present Datadog, Inc.
+ */
+
+package com.datadog.android.ktx.coroutine
+
+import com.datadog.android.rum.GlobalRum
+import com.datadog.android.rum.RumErrorSource
+import kotlinx.coroutines.flow.Flow
+import kotlinx.coroutines.flow.collect
+import kotlinx.coroutines.flow.flow
+
+internal const val ERROR_FLOW: String = "Coroutine Flow error"
+
+/**
+ *  Returns a [Flow] that will send a RUM Error event if this [Flow] emits an error.
+ *  Note that the error will also be emitted by the returned [Flow].
+ */
+@Suppress("TooGenericExceptionCaught")
+fun <T> Flow<T>.sendErrorToDatadog(): Flow<T> {
+    return flow {
+        try {
+            collect { value -> emit(value) }
```

Comment:
this means collecting each data from prev `Flow` and passing it to another one, I guess if the flow is big enough this can be heavy operation 🤔

Reply:
It won't be heavy. The `suspend` lambda passed into the collect is called with each new item of the stream. This means that this coroutine code will get awakened when a new value (or an exception) is emitted, and will forward that to the original collector. So it will never "hold" the collection of all items. 

`collect` ≠ `toList`